pkg/sdk/resource_monitors.go

```diff
@@ -47,9 +173,18 @@ func (v *ResourceMonitor) ObjectType() ObjectType {
 
 // CreateResourceMonitorOptions contains options for creating a resource monitor.
 type CreateResourceMonitorOptions struct {
-	create          bool                    `ddl:"static" sql:"CREATE"`           //lint:ignore U1000 This is used in the ddl tag
+	create          bool                    `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace       *bool                   `ddl:"keyword" sql:"OR REPLACE"`
 	resourceMonitor bool                    `ddl:"static" sql:"RESOURCE MONITOR"` //lint:ignore U1000 This is used in the ddl tag
 	name            AccountObjectIdentifier `ddl:"identifier"`
+	With            *bool                   `ddl:"keyword" sql:"WITH"`
+
+	CreditQuota    *int                 `ddl:"parameter,equals" sql:"CREDIT_QUOTA"`
+	Frequency      *Frequency           `ddl:"parameter,equals" sql:"FREQUENCY"`
+	StartTimeStamp *string              `ddl:"parameter,equals,single_quotes" sql:"START_TIMESTAMP"`
```

Comment:
minor point but Timestamp is typically written as one word. so this should be `StartTimestamp` and `EndTimestamp`

Reply:
fixed

Reply:
fixed